### PR TITLE
fix(curriculum) - Fixed regex parser for detecting h2 element

### LINKED ISSUE #38932: Wrong CSS selector type passes challenge
The challenge is asking to use an element selector but the regex allows for a class selector.

Challenge: https://www.freecodecamp.org/learn/responsive-web-design/basic-css/import-a-google-font

Forum post: https://www.freecodecamp.org/forum/t/html-css-font-help/395209

### PR DESCRIPTION
**Pull request by ptkpyitheim**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38932

Before:
Couldn't parse the h2 CSS selector because of an incorrect regex parser

After:
Can parse h2 selector correctly and also allows single and double quotes or without quotes for Lobster font

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ tests:
   - text: Your <code>h2</code> element should use the font <code>Lobster</code>.
     testString: assert($("h2").css("font-family").match(/lobster/i));
   - text: You should use an <code>h2</code> CSS selector to change the font.
-    testString: 'assert(/\s*h2\s*\{\s*font-family\:\s*(\"|")?Lobster(\"|")?(.{0,})\s*;\s*\}/gi.test(code));'
+    testString: assert(/\s*h2\s*\{\s*font-family\:\s*(\"|"|\')?Lobster(\"|"|\')?(.{0,})\s*;\s*\}/gi.test(code));
```
Comment (Sky020) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md:37]:
```suggestion
    testString: assert(/\s*h2\s*\{\s*font-family\:\s*(['"])?Lobster\1\s*;\s*\}/gi.test(code));
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Perhaps just a more standard way of testing for quotations

File: curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md (hunk 1 above)
Comment (ptkpyitheim) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md:37]: Great suggestion. Will make the changes!

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Sorry to crash the party, but the current challenge instructions do not state the user has to use an element selector and definitely does not say you can not use a class selector.  The only place that states this is in the 3rd test's `text`. I propose we should also modify the instructions to something like:
> Use an `h2` element selector to create a font-family CSS rule that uses the Lobster font to ensure it will be applied to your `h2` element.

**Comment by lasjorg** [comment]
@ptkpyitheim I think the proposed change to the challenge instructions would be nice to have. Would you mind updating the Instructions text with what @RandellDawson posted?

File: curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ tests:
   - text: Your <code>h2</code> element should use the font <code>Lobster</code>.
     testString: assert($("h2").css("font-family").match(/lobster/i));
   - text: You should use an <code>h2</code> CSS selector to change the font.
-    testString: 'assert(/\s*h2\s*\{\s*font-family\:\s*(\"|")?Lobster(\"|")?(.{0,})\s*;\s*\}/gi.test(code));'
+    testString: assert(/\s*h2\s*\{\s*font-family\:\s*(['"])?Lobster\1\s*;\s*\}/gi.test(code));
```
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.english.md:37]:
```suggestion
    testString: assert(/\s*h2\s*\{\s*font-family\:\s*(['"])?Lobster\1?\s*(;\s*\}|\})/gi.test(code));
```

The current changes don't allow for no quotes. And the stuff I added at the end allow users to omit the semi-colon if it's followed by the closing `}` - which is valid css.

**Review by moT01 — CHANGES_REQUESTED** [review]
I also like @RandellDawson's suggestion for the instructions, but the challenge also needs users to import the font which is not mentioned in the instructions.

How about...
`
Import the `Lobster` font to your web page. Then use an h2 element selector to create a font-family CSS rule that uses the Lobster font to ensure it will be applied to your h2 element.
`

**Comment by moT01** [comment]
Thanks for the PR, but we have decided to go with an ealier PR that makes a fix for this issue. Thanks and happy coding! 🎉

**Comment by ptkpyitheim** [comment]
I see that the PR has been merged. Do I still need to make any changes?

**Comment by RandellDawson** [comment]
> I see that the PR has been merged. Do I still need to make any changes?

No, everything should be good to go based on the other PR merged.  Feel free to tackle any of the other outstanding issues though.  We would welcome future PRs from you.

**Comment by ptkpyitheim** [comment]
> > I see that the PR has been merged. Do I still need to make any changes?
> 
> No, everything should be good to go based on the other PR merged. Feel free to tackle any of the other outstanding issues though. We would welcome future PRs from you.

Thank you I appreciate it.

**Pull request closed without merging**